pkg/multicast/multicast.go

```diff
@@ -18,8 +21,9 @@ import (
 )
 
 const (
-	defaultReadBufferSize = 1500
-	defaultDataChSize     = 1000
+	defaultReadBufferSize   = 1500
+	defaultDataChSize       = 1000
+	defaultEventQueueLength = 100
```

Comment:
Update to `1000`

Reply:
Still get this error event the queue length is `1000`:

```
2022-08-02T11:52:10.278Z        error   multicast/multicast.go:504      failed to insert data to event queue    {"addr": "239.111.111.2:6100", "err": "offset out of range", "offset": 1001, "channelID": 2, "seq": 597073949}                                                                                                      
2022-08-02T11:52:10.278Z        info    multicast/multicast.go:597      restarting connection...        {"addr": "239.111.111.2:6100", "error": "connection reset"}                                                                                                                                                                 
2022-08-02T11:52:10.278Z        info    multicast/multicast.go:615      connection closed       {"addr": "239.111.111.2:6100", "error": "read udp 0.0.0.0:6100: us
e of closed network connection"}
```

Reply:
1 package was lost util the offset is too large (package with sequence number 598704187):
```
2022-08-02T13:31:02.498Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704180}                                                                                         
2022-08-02T13:31:02.551Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704181}                                                                                         
2022-08-02T13:31:02.558Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704182}                                                                                         
2022-08-02T13:31:02.583Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704183}                                                                                         
2022-08-02T13:31:02.584Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704184}                                                                                         
2022-08-02T13:31:02.586Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704185}                                                                                         
2022-08-02T13:31:02.586Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704186}                                                                                         
2022-08-02T13:31:02.586Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704188}                                                                                         
2022-08-02T13:31:02.589Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704189}                                                                                         
2022-08-02T13:31:02.596Z        debug   multicast/multicast.go:450      Handle package  {"channelID": 2, "seq": 598704190}
```

Reply:
Set `defaultReadBufferSize = 15000` resolve the problem.

Reply:
Set `defaultReadBufferSize = 30000`, packages was lost after 1m46s:

```
2022-08-02T14:15:56.977Z        info    websocket/client.go:146 auto reconnect is enable        {"func": "start"}                                                                                                  
2022-08-02T14:17:12.677Z        error   multicast/multicast.go:490      Out of order package    {"addr": "239.111.111.2:6100", "channelID": 2, "offset": 9, "seq": 600002338}                                      
2022-08-02T14:17:12.677Z        error   multicast/multicast.go:490      Out of order package    {"addr": "239.111.111.2:6100", "channelID": 2, "offset": 10, "seq": 600002339}                                     
2022-08-02T14:17:12.677Z        error   multicast/multicast.go:490      Out of order package    {"addr": "239.111.111.2:6100", "channelID": 2, "offset": 11, "seq": 600002340}
```

Reply:
Set `defaultReadBufferSize = 65535`, the program run fine with one address in 10m.

Reply:
There was a lot of duplicated packages when run with 2 addresses:
```
2022-08-02T14:50:38.669Z        debug   multicast/multicast.go:465      readPackageHeader: duplicated package   {"last_sequence": 601197955, "current_sequence": 601197955, "channel_id": 2}
2022-08-02T14:50:38.669Z        debug   multicast/multicast.go:483      Duplicate package       {"addr": "239.111.111.2:6100", "channelID": 2, "offset": 0, "seq": 601197955}
2022-08-02T14:50:38.670Z        debug   multicast/multicast.go:465      readPackageHeader: duplicated package   {"last_sequence": 601197957, "current_sequence": 601197957, "channel_id": 2}
2022-08-02T14:50:38.670Z        debug   multicast/multicast.go:483      Duplicate package       {"addr": "239.111.111.3:6100", "channelID": 2, "offset": 0, "seq": 601197957}
2022-08-02T14:50:38.670Z        debug   multicast/multicast.go:465      readPackageHeader: duplicated package   {"last_sequence": 601197957, "current_sequence": 601197957, "channel_id": 2}
2022-08-02T14:50:38.670Z        debug   multicast/multicast.go:483      Duplicate package       {"addr": "239.111.111.2:6100", "channelID": 2, "offset": 0, "seq": 601197957}
2022-08-02T14:50:38.671Z        debug   multicast/multicast.go:465      readPackageHeader: duplicated package   {"last_sequence": 601197958, "current_sequence": 601197958, "channel_id": 2}
2022-08-02T14:50:38.671Z        debug   multicast/multicast.go:483      Duplicate package       {"addr": "239.111.111.3:6100", "channelID": 2, "offset": 0, "seq": 601197958}
```

Reply:
```
2022-08-02T14:58:04.919Z        error   multicast/multicast.go:490      Out of order package    {"addr": "239.111.111.2:6100", "channelID": 2, "offset": 3693468426, "seq": 601498872}
2022-08-02T14:58:04.919Z        error   multicast/multicast.go:490      Out of order package    {"addr": "239.111.111.3:6100", "channelID": 2, "offset": 3693468426, "seq": 601498872}
2022-08-02T14:58:04.919Z        error   multicast/multicast.go:506      failed to insert data to event queue    {"addr": "239.111.111.2:6100", "err": "offset out of range", "offset": 3693468426, "channelID": 2, "seq": 601498872}
2022-08-02T14:58:04.919Z        error   multicast/multicast.go:506      failed to insert data to event queue    {"addr": "239.111.111.3:6100", "err": "offset out of range", "offset": 3693468426, "channelID": 2, "seq": 601498872}
2022-08-02T14:58:04.919Z        info    multicast/multicast.go:599      restarting connection...        {"addr": "239.111.111.2:6100", "error": "connection reset"}
2022-08-02T14:58:04.919Z        info    multicast/multicast.go:599      restarting connection...        {"addr": "239.111.111.3:6100", "error": "connection reset"}
```

Reply:
2 connections always handle the same package at a time:
```
2022-08-02T15:18:54.060Z        debug   multicast/multicast.go:626      Package header  {"addr": "239.111.111.3:6100", "len": 85, "channelID": 3, "seq": 364490994, "error": null}
2022-08-02T15:18:54.060Z        debug   multicast/multicast.go:626      Package header  {"addr": "239.111.111.2:6100", "len": 85, "channelID": 3, "seq": 364490994, "error": null}
2022-08-02T15:18:54.062Z        debug   multicast/multicast.go:626      Package header  {"addr": "239.111.111.3:6100", "len": 85, "channelID": 3, "seq": 364490995, "error": null}
2022-08-02T15:18:54.062Z        debug   multicast/multicast.go:626      Package header  {"addr": "239.111.111.2:6100", "len": 85, "channelID": 3, "seq": 364490995, "error": null}
2022-08-02T15:18:54.062Z        debug   multicast/multicast.go:465      readPackageHeader: duplicated package   {"last_sequence": 364490995, "current_sequence": 364490995, "channel_id": 3}
2022-08-02T15:18:54.062Z        debug   multicast/multicast.go:465      readPackageHeader: duplicated package   {"last_sequence": 364490995, "current_sequence": 364490995, "channel_id": 3}
2022-08-02T15:18:54.062Z        debug   multicast/multicast.go:482      Duplicate package       {"addr": "239.111.111.2:6100", "channelID": 0, "offset": 0, "seq": 0}
2022-08-02T15:18:54.062Z        debug   multicast/multicast.go:482      Duplicate package       {"addr": "239.111.111.3:6100", "channelID": 0, "offset": 0, "seq": 0}
```

Reply:
https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/net/sock_posix.go;drc=f2a9f3e2e0ce7e582d226ad9a41d3c36b146fc25;l=207

Reply:
Listen to a list of multicast addresses with same port using ipv4 package: https://pkg.go.dev/golang.org/x/net/ipv4#hdr-Multicasting